apis/datadoghq/v1alpha1/datadogautoscaling_types.go

```diff
@@ -0,0 +1,282 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package v1alpha1
+
+import (
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// spec:
+//   targetRef:
+//     apiVersion: apps/v1
+//     kind: Deployment
+//     name: test
+//   owner: local
+//   policy:
+//     update:
+//       mode: Auto|DryRun
+//   recommender:
+//     name: containerMetrics
```

Comment:
Maybe it sounds a bit generic, should we have something that is more self-describing about what's the recommender logic is? Seems we're headed towards multidimensional with watermarks.

Reply:
Yes that's something we need to clearly define, this is more of an example.